pkg/satellite/satelliteweb/satelliteql/mutation_test.go

```diff
@@ -0,0 +1,489 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satelliteql
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/graphql-go/graphql"
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/stretchr/testify/assert"
+
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/pkg/auth"
+	"storj.io/storj/pkg/satellite"
+	"storj.io/storj/pkg/satellite/satelliteauth"
+	"storj.io/storj/pkg/satellite/satellitedb"
+
+	"testing"
+)
+
+func TestGrapqhlMutation(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	log := zap.NewExample()
+
+	db, err := satellitedb.New("sqlite3", "file::memory:?mode=memory&cache=shared")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = db.CreateTables(); err != nil {
+		t.Fatal(err)
+	}
+
+	service, err := satellite.NewService(
+		log,
+		&satelliteauth.Hmac{Secret: []byte("my-suppa-secret-key")},
+		db,
+	)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	creator := TypeCreator{}
+	if err = creator.Create(service); err != nil {
+		t.Fatal(err)
+	}
+
+	schema, err := graphql.NewSchema(graphql.SchemaConfig{
+		Query:    creator.RootQuery(),
+		Mutation: creator.RootMutation(),
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	createUser := satellite.CreateUser{
+		UserInfo: satellite.UserInfo{
+			FirstName: "John",
+			LastName:  "Roll",
+			Email:     "test@email.com",
+		},
+		Password: "123a123",
+	}
+
+	rootUser, err := service.CreateUser(ctx, createUser)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	token, err := service.Token(ctx, createUser.Email, createUser.Password)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sauth, err := service.Authorize(auth.WithAPIKey(ctx, []byte(token)))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	authCtx := satellite.WithAuth(ctx, sauth)
+
+	t.Run("Create user mutation", func(t *testing.T) {
+		newUser := satellite.CreateUser{
+			UserInfo: satellite.UserInfo{
+				FirstName: "Mickey",
+				LastName:  "Green",
+				Email:     "u1@email.com",
+			},
+			Password: "123a123",
+		}
+
+		query := fmt.Sprintf(
+			"mutation {createUser(input:{email:\"%s\",password:\"%s\",firstName:\"%s\",lastName:\"%s\"})}",
```

Comment:
`%s` doesn't escape the string properly, maybe use `%q` if the syntax is compatible with the graphql format, otherwise use an appropriate escaping func. (same for other places)